compiler/noirc_evaluator/src/ssa/opt/pure.rs

```diff
@@ -271,6 +280,32 @@ fn analyze_call_graph(
 
     finished_purities
 }
+
+/// Returns the set of Brillig functions that are entrypoints:
+/// main (if it's Brillig) and any Brillig functions that are called from ACIR.
+fn compute_brillig_entry_points(ssa: &Ssa) -> HashSet<FunctionId> {
```

Comment:
We actually have a similar method in the `brillig_entry_points` pass. It makes a mapping of entry points -> internal calls, but we could just call that method and fetch the keys of the map.